docusaurus/README.md

```diff
@@ -18,26 +18,63 @@ yarn install
 yarn build
 ```
 
-At this point you will see any broken links that docusaurus was able to find.
+`yarn build` will build Docusaurus site in `docusaurus/build` directory.
 
 ## Developing Locally
 
+If you want to make changes to the documentation, you can run docusaurus locally in a way that would
+listen to any source docs changes and live-reload them:
+
 ```bash
 yarn start # any changes will automatically be reflected in your browser!
 ```
 
-## Making Changes
+All the content for docs.airbyte.com lives in the `/docs` directory in this repo. All files are
+markdown. Make changes or add new files, and you should see them in your browser!
+
+## Changing Navigation Structure
+
+If you have created any new files, be sure to add them manually to the table of contents found here
+in [`sidebars.js`](https://github.com/airbytehq/airbyte/blob/master/docusaurus/sidebars.js)
+
+## Contributing
+
+We welcome documentation updates! If you'd like to contribute a change, please make sure to:
 
-All the content for docs.airbyte.com lives in the `/docs` directory in this repo. All files are markdown. Make changes or add new files, and you should see them in your browser!
+- Run `yarn build` and check that all build steps are successful.
+- Push your changes into a pull request, and follow the PR template instructions.
 
-If you have created any new files, be sure to add them manually to the table of contents found here in this [file](https://github.com/airbytehq/airbyte/blob/master/docusaurus/sidebars.js)
+When you make a pull request, Vercel will automatically build a test instance of the full docs site
+and link it in the pull request for review.
 
-## Plugin Client Redirects
+### Checking for broken links
+
+Airbyte docs site checks link with Docusaurus at build time, and with an additional GitHub action
+periodically:
+
+- Running the build process will **check for broken links**, please read the output and address any
+  broken links that you are able to do.
+- [This GitHub Action](https://github.com/airbytehq/airbyte/blob/master/.github/workflows/doc-link-check.yml)
+  checks all links on Airbyte production docs site, and tells us if any of them are broken.
+
+> [!NOTE] Docusaurus links checker only checks _relative_ links, and assumes that absolute links are
+> fine. For that reason, if you're linking to another Airbyte documentation page, make it a relative
+> link. I.e. `[link](/connector-development/overview.md)` instead of
+> `[link](https://docs.airbyte.com/connector-development/)`. That way, if your link breaks in the
+> future, it will be caught with `yarn build`.
```

Comment:
```suggestion
> `[link](https://docs.airbyte.com/connector-development/)`. That way, if your link breaks in the
> future due to a navigation restructure, it will be caught with `yarn build`.
```
probably the most likely reason